modules/lvm2/udiskslinuxvdovolume.c

```diff
@@ -281,11 +304,138 @@ handle_enable_deduplication (UDisksVDOVolume       *_volume,
                                            FALSE /* compression */, TRUE /* deduplication */, options);
 }
 
+static gboolean _vdo_resize (UDisksLinuxLogicalVolumeObject *object,
+                             GDBusMethodInvocation          *invocation,
+                             guint64                         new_size,
+                             GVariant                       *options)
+{
+  GError *error = NULL;
+  UDisksDaemon *daemon = NULL;
+  UDisksLinuxVolumeGroupObject *group_object = NULL;
+  uid_t caller_uid;
+  LVJobData data;
+  gboolean success = FALSE;
+
+  if (!common_setup (object, invocation, options,
+                     N_("Authentication is required to resize a VDO volume"),
+                     &daemon, &caller_uid))
+    goto out;
+
+  group_object = udisks_linux_logical_volume_object_get_volume_group (object);
+  data.vg_name = udisks_linux_volume_group_object_get_name (group_object);
+  data.lv_name = udisks_linux_logical_volume_object_get_name (object);
+  data.new_lv_size = new_size;
+
+  data.resize_fs = FALSE;
+  data.force = FALSE;
+  g_variant_lookup (options, "resize_fsys", "b", &(data.resize_fs));
+  g_variant_lookup (options, "force", "b", &(data.force));
+
+  if (!udisks_daemon_launch_threaded_job_sync (daemon,
+                                               UDISKS_OBJECT (object),
+                                               "lvm-lvol-resize",
+                                               caller_uid,
+                                               lvresize_job_func,
+                                               &data,
+                                               NULL, /* user_data_free_func */
+                                               NULL, /* GCancellable */
+                                               &error))
+    {
+      g_dbus_method_invocation_return_error (invocation,
+                                             UDISKS_ERROR,
+                                             UDISKS_ERROR_FAILED,
+                                             "Error resizing logical volume: %s",
+                                             error->message);
+      g_clear_error (&error);
+      goto out;
+    }
+
+  success = TRUE;
+
+ out:
+  g_clear_object (&object);
```

Comment:
This is a nonsense, you're taking pointer to a stack variable which you shouldn't even modify since you don't own the memory ownership.

Reply:
:man_facepalming: Too much copy-pasting, not enough thinking. Fixed and thank you for noticing this.